hooks/hookexecution/execution.go

```diff
@@ -311,3 +315,40 @@ func handleHookMutations[P any](
 
 	return payload
 }
+
+func handleModuleActivities[T any, P any](hook hooks.HookWrapper[T], activityControl privacy.ActivityControl, payload P) P {
+	// only 2 stages receive bidder request: hookstage.ProcessedAuctionRequestPayload and hookstage.BidderRequestPayload
+	// they both implement PayloadBidderRequest interface in order to execute mutations on bid request
+	var payloadData hookstage.PayloadBidderRequest
+	ok := true
```

Comment:
Nitpick: Might look a bit nicer to group the definitions.. or maybe not,. This one may be subjective. The `ok := true` just looks a bit odd to me (although I understand why it's done).

```
payloadData, ok := any(&payload).(hookstage.PayloadBidderRequest)
!ok {
  // payload doesn't have a bid request
  return payload
}
```